### PR TITLE
CFP Page

### PR DESCRIPTION
**Pull request by leerob**

### CODE REVIEW

File: components/conf/layout.module.css [hunk 1]
```diff
@@ -146,7 +146,7 @@
   margin-bottom: 15px;
 }
 
-@media screen and (min-width: 768px) {
+@media screen and (min-width: 800px) {
```
Comment (lfades) [inline components/conf/layout.module.css:149]: This change can produce unexpected results:

Live:

image

With this PR:

image

File: pages/conf/cfp.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { SkipNavContent } from '@reach/skip-nav';
+import { SITE_URL } from '@lib/constants';
+import Page from '@components/page';
+import SocialMeta from '@components/social-meta';
+import Layout from '@components/conf/layout';
+
+import Wizard from '@components/conf/wizard';
+
+export default function CallForPapers() {
```
Comment (lfades) [inline pages/conf/cfp.tsx:9]:
```suggestion
export default function CallForProposals() {
```

File: components/conf/wizard.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(values)
+      });
+
+      if (!res.ok) {
+        throw new Error('Failed to submit proposal.');
+      }
```
Comment (lfades) [inline components/conf/wizard.tsx:62]: Can we use the response from the API here to provide better details on what happened? If there's something useful that the user may need to know

File: components/conf/thank-you.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import cn from 'classnames';
+import styles from './thank-you.module.css';
+import styleUtils from './utils.module.css';
+
+export default function ThankYou() {
+  return (
+    <section className={cn(styles.section)}>
+      <div className={cn(styles.card, styleUtils.appear, styleUtils['appear-first'])}>
+        <h2 className={cn(styles.hero)}>Thank You</h2>
+        <p className={cn(styles.description)}>We'll review your submission and be in touch soon.</p>
```
Comment (lfades) [inline components/conf/thank-you.tsx:10]: You don't need to use `cn(...)` here

File: components/conf/layout.module.css (hunk 1 above)
Comment (lfades) [reply components/conf/layout.module.css:149]: Just noticed that this is expected as it's now including the email. But it still feels like a bad use of space 🤔

Comment (chibicode) [reply components/conf/layout.module.css:149]: Yeah it’s a hotfix to accommodate the email, I think it’s fine for now.

File: components/conf/thank-you.tsx (hunk 4 above)
Comment (chibicode) [reply components/conf/thank-you.tsx:10]: Yes, will clean this later

File: components/conf/wizard.tsx (hunk 3 above)
Comment (chibicode) [reply components/conf/wizard.tsx:61]: 👍

File: components/conf/layout.tsx [hunk 5]
```diff
@@ -65,8 +66,8 @@ export default function Layout({ children, inner, confLogoLink }: Props) {
           </div>
           <div
             className={cn(styles['header-right'], {
-              [styleUtils.appear]: !inner,
-              [styleUtils['appear-second']]: !inner
+              [styleUtils.appear]: !skipHeaderFooterAnimation,
```
Comment (leerob) [inline components/conf/layout.tsx:69]: Good refactor, much better name 💡

File: components/conf/layout.tsx [hunk 6]
```diff
@@ -106,7 +107,7 @@ export default function Layout({ children, inner, confLogoLink }: Props) {
             <div className={styles['footer-separator']} />
             <p className={styles['footer-paragraph']}>
               <a
-                href="https://www.notion.so/vercel/Next-js-Conf-Code-of-Conduct-2dae92927656409db28aaf2a62d99c41"
+                href={codeOfConductUrl}
```
Comment (leerob) [inline components/conf/layout.tsx:110]: Also definitely like everything pulled out into `/lib`.

File: components/conf/layout.tsx [hunk 7]
```diff
@@ -125,6 +126,28 @@ export default function Layout({ children, inner, confLogoLink }: Props) {
...
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                Proposal Guidelines
+              </a>
+            </p>
+            <div className={styles['footer-separator']} />
+            <p className={styles['footer-paragraph']}>
+              <a
+                href={`mailto:${confEmail}`}
```
Comment (leerob) [inline components/conf/layout.tsx:143]: Random thought: some people hate `mailto` links. Worth the extra UX to make this copy to clipboard instead?

File: components/conf/proposal-form-input.module.css [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  box-shadow: var(--geist-success-light) 0px 0px 0px 1px;
+  border-color: var(--geist-success-light);
+  background-color: #33373c;
+}
+
+.input:invalid {
+  box-shadow: none;
+}
+
+/* todo: do we want this?
```
Comment (leerob) [inline components/conf/proposal-form-input.module.css:34]: We can probably remove this.

File: components/conf/proposal-form-textarea.module.css [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+.textarea::placeholder {
+  color: var(--secondary-color);
+}
+
+.textarea:focus {
+  background-color: #33373c;
+}
+
+/* todo: same question about error state */
+.textarea:invalid {
```
Comment (leerob) [inline components/conf/proposal-form-textarea.module.css:23]: And here.

File: components/conf/proposal-form.tsx [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+                .catch(async err => {
+                  let message = 'Error! Please try again.';
+
+                  if (err instanceof FormError) {
+                    const { res } = err;
+                    const data = res.headers.get('Content-Type')?.includes('application/json')
+                      ? await res.json()
+                      : null;
+
+                    if (data?.error?.code === 'bad_email') {
```
Comment (leerob) [inline components/conf/proposal-form.tsx:90]: I'm assuming this is thrown by the underlying webhook?

File: components/conf/proposal-form.tsx [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+              </div>
+            </div>
+            {proposalType === 'speaker' && (
+              <div className={styles['form-group']}>
+                <div className={styles['small-heading']}>
+                  Lightning Talk{' '}
+                  <span className={styles['small-heading-optional']}>(Optional)</span>
+                </div>
+                <ProposalFormCheckbox
+                  labelProps={{ htmlFor: 'lightning-talk' }}
```
Comment (leerob) [inline components/conf/proposal-form.tsx:249]: Is there an a11y reason we went with `aria-label` for input / textarea but opted for a `label` for the checkbox?

**Comment by leerob** [comment]
A few small things:

- The `textarea` needs hover / focus states
- Do we need to send focus to the error after hitting submit? Wondering what that experience is currently like with a screen reader.
- The placeholder color is different on Firefox, but not Chrome / Safari 🤔 

image

File: components/conf/proposal-form.tsx [hunk 12]
```diff
@@ -0,0 +1,330 @@
...
+                  let message = 'Error! Please try again.';
+
+                  if (err instanceof FormError) {
+                    const { res } = err;
+                    const data = res.headers.get('Content-Type')?.includes('application/json')
+                      ? await res.json()
+                      : null;
+
+                    if (data?.error?.code === 'bad_email') {
+                      message = 'Please enter a valid email';
```
Comment (leerob) [inline components/conf/proposal-form.tsx:91]:
```suggestion
                      message = 'Please enter a valid email.';
```

File: components/conf/proposal-form.tsx (hunk 11 above)
Comment (chibicode) [reply components/conf/proposal-form.tsx:249]: @leerob I thought that would allow you to click on the label to select the check box :)

**Comment by chibicode** [comment]
@leerob we only change the background on `:focus` now for the text input used on the conference home page as well (which was the design decision @evilrabbit made I think), so `textarea` should be fine as is (I removed blue border just now: bd1d300)

https://nextjs.org/conf

File: components/conf/proposal-form.tsx (hunk 11 above)
Comment (leerob) [reply components/conf/proposal-form.tsx:249]: Yep, I see that now. Great idea!

File: components/conf/proposal-form.tsx (hunk 10 above)
Comment (leerob) [reply components/conf/proposal-form.tsx:90]: Saw your message on Slack. Confirmed.

**Comment by leerob** [comment]
My other comments are very minor. This is looking great, and ready to ship if necessary. 

+1 Should we open it up for review?

**Comment by leerob** [comment]
### Firefox Select Width

#### Before
image

#### After
image


### Firefox Placeholder Opacity

#### Before
image

#### After
image

File: components/conf/layout.tsx (hunk 7 above)
Comment (lfades) [reply components/conf/layout.tsx:143]: Nope!

Comment (leerob) [reply components/conf/layout.tsx:143]: 👌

**Review by chibicode — APPROVED** [review]

**Review by lfades — APPROVED** [review]

**Pull request merged**